pkg/reconciler/pubsubsource/resources/receive_adapter.go

```diff
@@ -42,10 +42,27 @@ const (
 	credsMountPath = "/var/secrets/google"
 )
 
+// DefaultSecretSelector is the default secret selector used to load the creds
+// for the receive adapter to auth with GCP.
+func DefaultSecretSelector() *corev1.SecretKeySelector {
+	return &corev1.SecretKeySelector{
+		LocalObjectReference: corev1.LocalObjectReference{
+			Name: "google-cloud-key",
+		},
+		Key: "key.json",
+	}
```

Comment:
defaults should be populated by webhook

Reply:
However, omission of secrets should be used to mean "use GCE metadata".

Reply:
I am going to hold off on this until I understand how the GCE metadata server will give this.